### PR TITLE
Feature/anshul birla/88 authenticated api

### LINKED ISSUE #88: Authenticated API + Refactoring before deployment

### PR DESCRIPTION
**Pull request by Anshul-Birla**
### Administrative Info

Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Github Issue]-[3-4 word description separated by dashes]`.

### What issue(s) does this branch close?

Closes #88 

### Changes

What changes did you make?

- Added support for authenticated API routes as well as dynamically updating security tokens 
- Added support for getting current users from storage instead of the api
- Refactored directories to be more organized
- Added support for profile images 

### Testing
- Waited one hour for APi token to expire, then navigated the app and ensured no 401 errors were being produced
- Added images to profiles and ensured they were being rendered correctly 
### Confirmation of Change

- Do the above to test changes

### CODE REVIEW

File: components/Profile/AdminTeacherProfile/ProfileViewLeft.tsx [hunk 1]
```diff
@@ -1,31 +1,169 @@
...
 
+type ProfilePictureProps = {
+  profileEditable: boolean;
+  onImageChange: (img: File) => void;
+  firstName: string;
+  lastName: string;
+  image: File | null;
+};
+
+const ProfilePicture: React.FC<ProfilePictureProps> = ({
```
Comment (Benson272) [inline components/Profile/AdminTeacherProfile/ProfileViewLeft.tsx:25]: Maybe have some way to indicate that the pfp is clickable/editable after clicking the edit button

File: context/AuthContext.tsx [hunk 2]
```diff
@@ -48,13 +49,17 @@ const init: AuthState = {
   updateUser: () => {
     return new Promise<boolean>(() => false);
   },
+  // getNewRefreshToken: () => {
```
Comment (allisonbhavsar) [inline context/AuthContext.tsx:52]: Did you mean to leave the commented code in this file?

**Review by allisonbhavsar — CHANGES_REQUESTED** [review]
Looks great! Just left a few minor suggestions and was also wondering if the client cares about being able to remove a profile pic?

**Review by Benson272 — APPROVED** [review]
lgtm

File: components/Profile/AdminTeacherProfile/ProfileViewRight.module.css [hunk 3]
```diff
@@ -38,16 +38,17 @@
 
 .buttonContainer {
   padding: 20px;
-  display: flex;
-  align-items: center;
-  justify-content: space-between;
+  /*display: flex;*/
+  /*align-items: center;*/
+  /*justify-content: space-between;*/
```
Comment (allisonbhavsar) [inline components/Profile/AdminTeacherProfile/ProfileViewRight.module.css:43]: did you mean to leave this?

**Review by allisonbhavsar — APPROVED** [review]
LGTM, just left one comment about commented out css styles

File: components/Profile/AdminTeacherProfile/ProfileViewRight.module.css (hunk 3 above)
Comment (Anshul-Birla) [reply components/Profile/AdminTeacherProfile/ProfileViewRight.module.css]: ah good check ty

**Pull request merged**